### ISSUE TITLE
Jail respawn teleport causing EventExceptions.

### ISSUE THREAD

**Issue by mibby** [EssentialsX/Essentials#621]
EssentialsX dev 342.

```
[00:43:06] [Server thread/ERROR]: Could not pass event PlayerRespawnEvent to WorldGuard v6.1.2-SNAPSHOT;no_git_id with custom flags plugin.
org.bukkit.event.EventException
    at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:46) ~[patched_1.9.4.jar:git-Paper-748]
    at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) [patched_1.9.4.jar:git-Paper-748]
    at com.earth2me.essentials.Teleport.respawnNow(Teleport.java:245) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.Teleport.respawn(Teleport.java:223) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.User.checkJailTimeout(User.java:469) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.EssentialsTimer.run(EssentialsTimer.java:87) [EssentialsX-2.0.1.jar:?]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftTask.run(CraftTask.java:58) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:352) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.D(MinecraftServer.java:795) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.DedicatedServer.D(DedicatedServer.java:403) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.C(MinecraftServer.java:723) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.run(MinecraftServer.java:622) [patched_1.9.4.jar:git-Paper-748]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.lang.NullPointerException
    at com.sk89q.worldguard.bukkit.ConfigurationManager.get(ConfigurationManager.java:222) ~[?:?]
    at com.sk89q.worldguard.bukkit.RegionQuery.getApplicableRegions(RegionQuery.java:94) ~[?:?]
    at com.sk89q.worldguard.bukkit.listener.WorldGuardPlayerListener.onPlayerRespawn(WorldGuardPlayerListener.java:313) ~[?:?]
    at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor421.execute(Unknown Source) ~[?:?]
    at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:44) ~[patched_1.9.4.jar:git-Paper-748]
    ... 15 more
[00:43:06] [Server thread/ERROR]: Could not pass event PlayerRespawnEvent to mcMMO v1.5.05-SNAPSHOT-b136
org.bukkit.event.EventException
    at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:46) ~[patched_1.9.4.jar:git-Paper-748]
    at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) [patched_1.9.4.jar:git-Paper-748]
    at com.earth2me.essentials.Teleport.respawnNow(Teleport.java:245) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.Teleport.respawn(Teleport.java:223) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.User.checkJailTimeout(User.java:469) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.EssentialsTimer.run(EssentialsTimer.java:87) [EssentialsX-2.0.1.jar:?]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftTask.run(CraftTask.java:58) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:352) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.D(MinecraftServer.java:795) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.DedicatedServer.D(DedicatedServer.java:403) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.C(MinecraftServer.java:723) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.run(MinecraftServer.java:622) [patched_1.9.4.jar:git-Paper-748]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.lang.UnsupportedOperationException: Not supported yet.
    at com.earth2me.essentials.OfflinePlayer.hasMetadata(OfflinePlayer.java:923) ~[?:?]
    at com.gmail.nossr50.util.player.UserManager.hasPlayerDataKey(UserManager.java:112) ~[?:?]
    at com.gmail.nossr50.listeners.PlayerListener.onPlayerRespawn(PlayerListener.java:410) ~[?:?]
    at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor25.execute(Unknown Source) ~[?:?]
    at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:44) ~[patched_1.9.4.jar:git-Paper-748]
    ... 15 more
[00:43:06] [Server thread/ERROR]: Could not pass event PlayerRespawnEvent to NoCheatPlus v3.14.0-SNAPSHOT-sMD5NET-b973
org.bukkit.event.EventException
    at org.bukkit.plugin.EventExecutor$1.execute(EventExecutor.java:46) ~[patched_1.9.4.jar:git-Paper-748]
    at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:517) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:502) [patched_1.9.4.jar:git-Paper-748]
    at com.earth2me.essentials.Teleport.respawnNow(Teleport.java:245) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.Teleport.respawn(Teleport.java:223) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.User.checkJailTimeout(User.java:469) [EssentialsX-2.0.1.jar:?]
    at com.earth2me.essentials.EssentialsTimer.run(EssentialsTimer.java:87) [EssentialsX-2.0.1.jar:?]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftTask.run(CraftTask.java:58) [patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:352) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.D(MinecraftServer.java:795) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.DedicatedServer.D(DedicatedServer.java:403) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.C(MinecraftServer.java:723) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.run(MinecraftServer.java:622) [patched_1.9.4.jar:git-Paper-748]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.lang.ClassCastException
[00:43:07] [Server thread/WARN]: [BetterCombat] Task #5350493 for BetterCombat v1.9 generated an exception
java.lang.UnsupportedOperationException: Not supported yet.
    at com.earth2me.essentials.OfflinePlayer.getAttribute(OfflinePlayer.java:1438) ~[?:?]
    at me.anderson.bc.Players.PlayerRespawn$1.run(PlayerRespawn.java:24) ~[?:?]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftTask.run(CraftTask.java:58) ~[patched_1.9.4.jar:git-Paper-748]
    at org.bukkit.craftbukkit.v1_9_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:352) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.D(MinecraftServer.java:795) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.DedicatedServer.D(DedicatedServer.java:403) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.C(MinecraftServer.java:723) [patched_1.9.4.jar:git-Paper-748]
    at net.minecraft.server.v1_9_R2.MinecraftServer.run(MinecraftServer.java:622) [patched_1.9.4.jar:git-Paper-748]
    at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
```

**Comment by SupaHam** [comment]
Apologies for this issue. This has been fixed in ac4a156. You can grab the fix from [build 357](https://ci.drtshock.net/job/EssentialsX/357/) on the CI server.

**Comment by mibby** [comment]
Thanks. :)
